fix: copy the current formats when insertings a character

jrmd opened the pull request:
### Description of the Change
Previously we would just insert the character at the point, however, when a character was inserted within a text format they would split the format and insert the character with no formats.

With this commit we copy the current `value` and insert out character here which preserves the surrounding formatting

Old Behaviour

https://user-images.githubusercontent.com/7556775/221526499-2ab9ff1d-e0be-4620-aa93-297cb53f9ebc.mov



New Behaviour

https://user-images.githubusercontent.com/7556775/221526534-449a7639-f591-425c-a666-c8b4bd56be6e.mov


### How to test the Change
<!-- Please provide steps on how to test or validate that the change in this PR works as described. -->
Create some text with a text format (bold/italic) put cursor in the middle add a special character ensure that the format isnt split by the special character

### Changelog Entry
> Fixed - Maintain text formatting when inserting a character.


### Credits
@jrmd


### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

jrmd committed:
fix: copy the current formats when insertings a character

Previously we would just insert the character at the point, however,
when a character was inserted within a text format they would split the
format and insert the character with no formats.

With this commit we copy the current `value` and insert out character
here which preserves the surrounding formatting

Sidsector9 reviewed (changes requested):
Thank you very much for the PR!

I tested the changes and it has worked partially for me.

The formatting doesn't seem to work when the string is partially formatted.

Please watch the 2 videos below:

### Working

https://user-images.githubusercontent.com/17757960/221834034-021fbfba-64ab-4f76-bea9-a6760f2ce90c.mp4

### Not working


https://user-images.githubusercontent.com/17757960/221834097-71a0af92-58e4-4c08-8aea-67b6deb033d4.mp4


Let us know if you need any assistance with this PR.

jrmd committed:
fix: add the current format to the strings current value

Updates newValue.formats to be the length of the new text string by
duplicating the formats at value.start; this fixes an issue where it
breaks the remaining format if it is added within the middle

jrmd commented:
hey @Sidsector9 thanks for testing! I believe i have fixed this by copying the current positions formats :)

https://user-images.githubusercontent.com/7556775/221841294-1394ab05-23a8-468c-9689-431e26528d13.mov

jeffpaul commented:
@jrmd welcome to Insert Special Characters and thanks for the PR, this is definitely a nice improvement in maintaining formatting... many thanks for the contribution!

Sidsector9 commented on tests/bin/initialize.sh:
```diff
@@ -1,4 +1,5 @@
-npm run env run tests-cli "wp post create --post_type=page --post_title='Page with special characters' --post_status='publish'"
\ No newline at end of file
+npm run env run tests-cli "wp post create --post_type=page --post_title='Page with special characters' --post_status='publish'"
+npm run env run tests-cli "wp post create --post_type=page --post_title='Page with a format and special characters' --post_status='publish'"
```
Thanks for the changes! It is working for me now 👍 

Can you also elaborate why was this command added?

jrmd replied:
@Sidsector9 appologies, i was trying to write a new cypress test for the change however, cypress was not playing nice, i thought i cleaned this up!

jrmd replied:
fixed!

jrmd committed:
chore: remove unneeded script from tests/bin/initialize.sh

jeffpaul committed:
Merge branch 'develop' into fix/inserting-characters-within-format

Sidsector9 committed:
Merge branch 'develop' into fix/inserting-characters-within-format
